odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -261,7 +261,9 @@ def write_serializer(item):
             if field.type not in ('many2many', 'one2many') and field.compute is None:
                 for item in itertools.chain(src_partners, [dst_partner]):
                     if item[column]:
-                        if column in summable_fields and values.get(column):
+                        if field.type == 'reference':
+                            values[column] = f'{item[column]._name},{item[column].id}'
```

Comment:
Why not let its as a recordset? `write()` should handle this correctly. (and it will actually prevent a SQL request coming from the Reference.convert_to_cache().
```suggestion
                            values[column] = item[column]
```

Reply:
There is no reason, I just copied the same style as the write_serializer function did. I fixed it and pushed.